drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -666,9 +682,10 @@ def _mark_complete_and_publish_grade(self):
 
         # There's no going back from "completed" status to "incomplete"
         self.completed = self.completed or self._is_answer_correct() or not self.attempts_remain
-        grade = self._get_grade()
+        grade = self._calculate_grade()
         # ... and from higher grade to lower
-        if grade > self.grade:
+        current_grade = self._get_grade_if_set()
+        if current_grade is None or grade > current_grade:
```

Comment:
Hmmm.. Does this mean if the learner changes their answer so that it's different from the correct answer, they'll see their new incorrect answer, but their grade remains the same?  Won't that be confusing to the learner?  It's also inconsistent with CAPA problems.  FYI @sstack22.
(The sandbox is not responding currently so I can't verify this.)